src/app/header/header.component.ts

```diff
@@ -1,42 +1,100 @@
 import { AuthService } from '../services/auth.service';
 import { Router } from '@angular/router';
-import { MaterialModule } from '../material.module';
 import { LanguageService } from '../services/language.service';
+import { BadgeService } from '../services/BadgeService/badge.service';
+import { Badge } from '../badge/models/badgeModel';
+import { NotifyService } from '../badge/notify';
+import { trigger, state, style, transition, animate, keyframes } from '@angular/animations';
+import { BadgeComponent } from '../badge/badge.component';
+import { interval } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { MatSnackBar } from '@angular/material';
 
 @Component({
-    selector: 'app-header',
-    templateUrl: './header.component.html',
-    styleUrls: ['./header.component.scss'],
-    providers: [MaterialModule]
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+  animations: [
+    trigger('newBadge', [
+      state('beforeAnimation', style({
+        opacity: 1
+      })),
+      state('afterAnimation', style({
+        opacity: 1
+      })),
+      transition('beforeAnimation => afterAnimation', animate('2000ms', keyframes([
+        style({transform: ' translateX(+1000px)', offset: 0}),
+        style({transform: ' translateX(-1000px)', offset: 0.01}),
+        style({transform: ' rotateY(0deg) ', offset: 0.60}),
+        style({transform: 'rotateY(360deg) ', offset: 1})
+      ])))
+    ])
+  ]
 })
 
 export class HeaderComponent implements OnInit {
-    language: any;
-    @Output() toggleNavbar = new EventEmitter<any>();
-    @Output() openEditProfile = new EventEmitter<any> ();
+  public language: any;
+  public badge: Badge;
+  public state = 'beforeAnimation';
+  @Output() toggleNavbar = new EventEmitter<any>();
+  @Output() openEditProfile = new EventEmitter<any>();
+  private observable;
+  private subscription;
 
-    navbartoggle() {
-        this.toggleNavbar.emit();
-    }
+  constructor(private _authService: AuthService, private _router: Router, private _langService: LanguageService, private _badgeService: BadgeService,
+              private _notify: NotifyService, private _snackbar: MatSnackBar) {
+  }
 
-    constructor(private _authService: AuthService, private _router: Router,  private _langService: LanguageService) { }
+  navbartoggle() {
+    this.toggleNavbar.emit();
+  }
 
-    ngOnInit() {
-        this._langService.loadLanguage().subscribe(response => {
-            this.language = response.pcprepkit.common.header;
-        });
-     }
+  ngOnInit() {
+    this._langService.loadLanguage().subscribe(response => {
+      this.language = response.pcprepkit.common.header;
+    });
+    this.observable = interval(500)
+      .pipe(tap(() => this.loadBadge()));
+    this.subscription = this.observable.subscribe();
+    this._notify.notifyObservable$.subscribe(res => {
+      this.ngOnInit();
```

Comment:
This would be triggered when the user completes one stage. Calling ngOnInit() would ensure that new badge is displayed without refreshing the whole page again. I've tested it but doesn't seem to be causing problems.

Reply:
So ngOnInit is calling another ngOnInit, from the code I can't understand what is stopping this ngOnInit to keep calling another instance of ngOnInit till the ram is filled. 

Another question I have is what does this "$" in notifyObservable$ do ?

Reply:
@shreyans29  ngOnInit() is called only when our Observable emits values. That would happen only when User' badge has been changed. (i.e update the medal image in header)
Use of '$' sign is just a convention in naming Observables.

Reply:
got it